packages/salesforcedx-vscode-core/src/orgBrowser/metadataOutlineProvider.ts

```diff
@@ -84,10 +84,10 @@ export class MetadataOutlineProvider
         break;
       case NodeType.Folder:
       case NodeType.MetadataType:
-        let nodeType = NodeType.MetadataComponent;
+        let nodeType: NodeType = NodeType.MetadataComponent;
         if (TypeUtils.FOLDER_TYPES.has(element.fullName)) {
           nodeType = NodeType.Folder;
-        } else if (element.type === NodeType.Folder && element.fullName) {
+        } else if (element.parent && element.parent.fullName === 'CustomObject') {
```

Comment:
I am guessing 'CustomObject' has element.type as NodeType.Folder ?

Reply:
`Custom Objects` metadata type has element.type as NodeType.MetadataType. `Account` , an object listed under `Custom Objects` has element.type as NodeType.Folder.

Reply:
So this is the source of the bug and not just a refactor ?

Reply:
Yes! it's the source of the bug.